### ISSUE TITLE
"Unable to detect bundler spec under <gem path>"

### ISSUE THREAD

**Issue by seanstory** [jruby/warbler#536]
### Environment:
- rbenv 1.2.0
- jruby 9.3.3.0
- bundler 2.4.6
- warbler 2.0.5
- OSX 13.2

### Issue
When running `bundle exec warble` I get this error:
```
Unable to detect bundler spec under '/opt/homebrew/Cellar/rbenv/1.2.0/versions/jruby-9.3.3.0/lib/ruby/gems/shared/gems/bundler-2.4.6'' and its sub-dirs
```

This makes sense, as there's no gemspec file there, just a `exe/` dir with `bundle` and `bundler` executables. The gemspec for bundler lives in 

```
/opt/homebrew/Cellar/rbenv/1.2.0/versions/jruby-9.3.3.0/lib/ruby/gems/shared/specifications/default/bundler-2.4.6.gemspec 
```

### Workaround
I copied the gemspec to where warbler seems to be looking for it, and now it's working. However, I expect this will fall apart if/when I update my bundler gem.

```
cp /opt/homebrew/Cellar/rbenv/1.2.0/versions/jruby-9.3.3.0/lib/ruby/gems/shared/specifications/default/bundler-2.4.6.gemspec opt/homebrew/Cellar/rbenv/1.2.0/versions/jruby-9.3.3.0/lib/ruby/gems/shared/gems/bundler-2.4.6/bundler.gemspec
```

**Comment by olleolleolle** [comment]
@seanstory Yes, we have had valiant efforts to try to support Bundler 2.x. 

Thanks for your report!

**Comment by dolzenko** [comment]
So there is no fix for this at the moment?

**Comment by dometto** [comment]
While I can reproduce this error with `bundler` v2.3, I no longer can with `bundler` v2.5. (I did not try with 2.4).

**Comment by dub357** [comment]
FYI - this isn't a Bundler 2.x issue...its an issue when Bundler is installed as a 'default' gem, which is the case for newer JRubies.
If you install and use a version of bundler as a 'standard' gem (basically one that isn't the same as the one installed by default and gets installed in the typical location), then things will work properly because warbler will find the bundler.gemspec file.

The problematic warbler code is the following:
https://github.com/jruby/warbler/blob/5ce7ec0dbe6c0f43d671c3f4a88f110838791b1c/lib/warbler/traits/bundler.rb#L43-L56

I say 'problematic' because this code does not work when Bundler is a default gem.  This code here seems to have been introduced in https://github.com/jruby/warbler/commit/89978dd1246d979b8dbcaea49610047d898b26ed because when asked for dependencies, Bundler seems to report itself with specification that includes a 'fake' loaded_from attribute.  To work around that, this warbler code tries to find the 'real' loaded_from (the gemspec file location) so that bundler can be included in the archive its building.

It might be best to alter this code, detect if Bundler is a default gem and then simply skip it from being included in the archive...
